src/App.tsx

```diff
@@ -15,36 +23,90 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+const getPreparedGoods = (
+  sortType: SortType,
+  isReversed: boolean,
+): string[] => {
+  const goods = [...goodsFromServer];
+
+  goods.sort((a, b) => {
+    switch (sortType) {
+      case SortType.Alphabet:
+        return a.localeCompare(b);
+      case SortType.Length:
+        return a.length - b.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    goods.reverse();
+  }
+
+  return goods;
+};
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState<SortType>(SortType.Default);
+  const [isSortReversed, setSortIsReversed] = useState<boolean>(false);
+
+  const reset = () => {
+    setSortType(SortType.Default);
+    setSortIsReversed(false);
+  };
+
+  const isResetButtonVisible = sortType !== SortType.Default || isSortReversed;
+
+  const sortAndReversed = getPreparedGoods(sortType, isSortReversed);
```

Comment:
```suggestion
  const visibleGoods = getPreparedGoods(sortType, isSortReversed);
```

let's use more informative naming